backend/src/graphql/resolver/BalanceResolver.ts

```diff
@@ -5,36 +5,74 @@ import { Resolver, Query, Ctx, Authorized } from 'type-graphql'
 import { Balance } from '@model/Balance'
 import { calculateDecay } from '@/util/decay'
 import { RIGHTS } from '@/auth/RIGHTS'
-import { Transaction } from '@entity/Transaction'
+import { Transaction as dbTransaction } from '@entity/Transaction'
 import Decimal from 'decimal.js-light'
+import { GdtResolver } from './GdtResolver'
+import { TransactionLink as dbTransactionLink } from '@entity/TransactionLink'
+import { MoreThan, getCustomRepository } from '@dbTools/typeorm'
+import { TransactionLinkRepository } from '@repository/TransactionLink'
 
 @Resolver()
 export class BalanceResolver {
   @Authorized([RIGHTS.BALANCE])
   @Query(() => Balance)
   async balance(@Ctx() context: any): Promise<Balance> {
-    // load user and balance
     const { user } = context
     const now = new Date()
 
-    const lastTransaction = await Transaction.findOne(
-      { userId: user.id },
-      { order: { balanceDate: 'DESC' } },
-    )
+    const gdtResolver = new GdtResolver()
+    const balanceGDT = await gdtResolver.gdtBalance(context)
+
+    const lastTransaction = context.lastTransaction
+      ? context.lastTransaction
+      : await dbTransaction.findOne({ userId: user.id }, { order: { balanceDate: 'DESC' } })
 
     // No balance found
     if (!lastTransaction) {
       return new Balance({
         balance: new Decimal(0),
         decay: new Decimal(0),
-        decay_date: now.toString(),
+        lastBookedBalance: new Decimal(0),
+        balanceGDT,
+        count: 0,
+        linkCount: 0,
       })
     }
 
+    const count =
+      context.count || context.count === 0
+        ? context.count
+        : await dbTransaction.count({ where: { userId: user.id } })
+    const linkCount =
+      context.linkCount || context.linkCount === 0
+        ? context.linkCount
+        : await dbTransactionLink.count({
+            where: {
+              userId: user.id,
+              redeemedAt: null,
+              validUntil: MoreThan(new Date()),
+            },
+          })
+
+    const transactionLinkRepository = getCustomRepository(TransactionLinkRepository)
+    const { sumHoldAvailableAmount } = context.sumHoldAvailableAmount
+      ? { sumHoldAvailableAmount: context.sumHoldAvailableAmount }
+      : await transactionLinkRepository.summary(user.id, now)
```

Comment:
if you want to optimize queries, this query could return the `linkCount` aswell and the caching mechanism becomes obsolete.

Reply:
I use the context to pass the information that is already queried by the transaction resolver to avoid further queries here. This is not optimization as I already know the values.